### ISSUE TITLE
CoreML LayoutLM model predicts all O labels — zero labels reach DynamoDB

### ISSUE THREAD

**Issue by tnorlund** [tnorlund/Portfolio#809]
## Summary

The on-device CoreML LayoutLM model running in the Mac OCR Swift worker predicts `O` (outside/no label) for **every token** on **every receipt**. As a result, zero word labels are uploaded to DynamoDB from the Mac pipeline.

## Evidence

Verified across 3 recently uploaded images:

| Image | Receipt | Tokens | Non-O Predictions |
|-------|---------|--------|-------------------|
| `1828b9ba` | R1 (Vons) | 119 | **0** |
| `1828b9ba` | R2 (Sprouts) | 123 | **0** |
| `3f2d27b9` | R1 (JINYA) | 81 | **0** |

The OCR JSON on S3 (`s3://upload-images-image-bucket-4bcea7e/ocr_results/receipt-*.json`) contains `layoutlm_predictions` arrays where every entry has label `O`.

The Swift worker correctly guards against uploading empty labels:
```swift
// OCRWorker.swift ~line 309
guard let predictions = receipt.layoutLMPredictions, !predictions.isEmpty else { continue }
```

Since all predictions are `O`, `filteredLabels` is empty after filtering, so `upload_labels` is never called. This is confirmed by CloudWatch logs showing zero `upload_labels` log entries.

## Pipeline flow

```
Mac Swift Worker
  → VisionOCREngine runs LayoutLM CoreML inference
  → All predictions = O
  → OCR JSON written to S3 (predictions included but all O)
  → SQS → Python Lambda (ocr_processor.py)
  → Lambda creates Receipt/Line/Word/Letter entities
  → Lambda does NOT process layoutlm_predictions at all
  → Swift worker skips label upload (empty after filtering)
  → Result: 0 labels in DynamoDB
```

## Server-side inference works fine

The Lambda-based PyTorch LayoutLM inference (used by the batch cache generator) works correctly at ~78.8% average accuracy across 400+ cached receipts. This confirms the model weights and tokenizer are valid — the issue is specific to the CoreML export or the Swift preprocessing pipeline.

## Likely root causes to investigate

1. **CoreML model export** — the `.mlmodel` / `.mlpackage` may have been exported incorrectly (wrong output head, missing fine-tuned weights, or defaulting to base model)
2. **Tokenizer mismatch** — the Swift tokenizer may produce different token IDs than the Python tokenizer
3. **Label map mismatch** — the CoreML model's output indices may not map to the same labels as the Python model
4. **Input preprocessing** — bounding box normalization or token alignment may differ between Swift and Python
5. **Model file staleness** — the cached CoreML model may be from a pre-fine-tuning checkpoint

## Workaround

Labels are currently being added manually via MCP tools. The server-side batch cache generator (`layoutlm-batch-cache-dev-state-machine`) is working and can be triggered to regenerate inference cache for the frontend.
